#Q1
T-Test:
The t-test is used when the population standard deviation is unknown, and the sample size is small. It uses the t-distribution to account for the uncertainty introduced by estimating the population standard deviation from the sample. There are different types of t-tests, such as the independent samples t-test for comparing means of two independent groups and the paired samples t-test for comparing means of paired or matched observations.

Example Scenario for T-Test:
Suppose you want to test whether a new teaching method has a significant effect on student performance. You collect test scores from two groups: one group taught using the new method and another group taught using the traditional method. Since you don't know the population standard deviation of the test scores, and your sample sizes are relatively small, you would use an independent samples t-test to determine if there is a statistically significant difference in the means of the two groups.

Z-Test:
The z-test is used when the population standard deviation is known or when the sample size is large (typically n > 30). It assumes that the sample mean follows a normal distribution due to the Central Limit Theorem. Because of the larger sample size assumption, the z-test can be more powerful than the t-test.

Example Scenario for Z-Test:
Imagine you are assessing the effectiveness of a weight loss program. You have a large sample of participants (more than 30) and you know the population standard deviation of weight loss for this program. You want to determine if the average weight loss in your sample is significantly different from a claimed average weight loss of 5 kg. In this case, you could use a z-test to test the hypothesis and make a decision about whether the program's average weight loss is significantly different from the claimed value.

#Q2
One-Tailed Test:
A one-tailed test, also known as a directional test, is a type of hypothesis test where the null hypothesis is tested against an alternative hypothesis in only one direction. In other words, you are only interested in detecting a change or difference in one specific direction. The critical region for the test is located entirely on one side of the distribution, either the left side or the right side, depending on the specific hypothesis being tested.

There are two types of one-tailed tests:

Right-Tailed Test: In a right-tailed test, the critical region is in the upper tail of the distribution. This type of test is used when you are testing whether a sample statistic is significantly greater than a certain value.
Left-Tailed Test: In a left-tailed test, the critical region is in the lower tail of the distribution. This type of test is used when you are testing whether a sample statistic is significantly less than a certain value.
Two-Tailed Test:
A two-tailed test, also known as a non-directional test, is a type of hypothesis test where the null hypothesis is tested against an alternative hypothesis in both directions. In this case, you are interested in detecting a change or difference in either direction from the hypothesized value. The critical region for the test is divided between the two tails of the distribution.

Two-tailed tests are used when you want to test whether a sample statistic is significantly different from a certain value, without specifying whether it is greater or smaller. The critical region includes both extremes of the distribution.

Example Scenarios:

One-Tailed Test: Suppose you are testing a new drug and you expect it to increase participants' reaction times. You conduct a one-tailed test to determine if the drug's effect is statistically significant in increasing reaction times.

Two-Tailed Test: Consider a scenario where you are testing whether the average IQ score of a group of students is significantly different from a national average of 100. You are not sure if the group's IQ scores will be higher or lower, so you use a two-tailed test to capture differences in both directions.

#Q3
In hypothesis testing, Type I and Type II errors are two possible mistakes that can occur when making a decision about a null hypothesis. These errors are associated with the outcomes of hypothesis tests and have different implications in terms of the conclusions drawn from the test.

Type I Error (False Positive):
A Type I error occurs when the null hypothesis is rejected when it is actually true. In other words, it's a false alarm where you conclude that there is an effect or a difference when there is none in reality. The probability of committing a Type I error is denoted as the significance level (α), which is typically set before conducting the test.

Example Scenario - Type I Error:
Imagine you're testing a new drug's effectiveness in reducing cholesterol levels. The null hypothesis states that the drug has no effect on cholesterol levels. If you conduct a hypothesis test and reject the null hypothesis, claiming that the drug does indeed reduce cholesterol levels, but in reality, it has no effect, you have committed a Type I error. This could lead to unnecessary medical interventions and expenses based on incorrect conclusions.

Type II Error (False Negative):
A Type II error occurs when the null hypothesis is not rejected when it is actually false. In other words, you fail to detect an effect or a difference that does exist. The probability of committing a Type II error is denoted as β.

Example Scenario - Type II Error:
Continuing with the example of testing a new drug's effectiveness, consider that the drug does indeed reduce cholesterol levels. If you conduct a hypothesis test but fail to reject the null hypothesis, concluding that the drug has no effect, you have committed a Type II error. This could lead to missed opportunities for medical advancements and improvements in patient health.

#Q4
Bayes's Theorem is a fundamental concept in probability theory that describes how to update the probability of a hypothesis based on new evidence. It combines prior beliefs or knowledge with observed data to revise the probability estimate of a hypothesis.

The theorem is expressed as:
P(A∣B)= P(B∣A)⋅P(A)/P(B)

Where:
P(A∣B) is the probability of hypothesis A being true given evidence B.

P(B∣A) is the probability of observing evidence B given that hypothesis A is true.

P(A) is the prior probability of hypothesis A being true.

P(B) is the probability of observing evidence B.
Example Scenario: Medical Test
Let's illustrate Bayes's Theorem with a medical test example. Imagine you are using a diagnostic test to determine whether a patient has a certain rare disease (Disease A). The test result can be positive (indicating the presence of the disease) or negative (indicating the absence of the disease).

Given:

The prior probability that a randomly selected person has Disease A is 0.01 (1% of the population has the disease), so 

P(A)=0.01.
The test's sensitivity (true positive rate) is 0.95, meaning that 

P(B∣A)=0.95, i.e., the probability of testing positive given that the patient has the disease.
The test's specificity (true negative rate) is 0.90, meaning that 

P(¬B∣¬A)=0.90, i.e., the probability of testing negative given that the patient doesn't have the disease.
Now, let's calculate the probability that a person actually has the disease given that they tested positive, 

P(A∣B), using Bayes's Theorem.

P(A∣B)= P(B∣A)⋅P(A)/P(B)

First, calculate 

P(B), the probability of testing positive:

P(B)=P(B∣A)⋅P(A)+P(B∣¬A)⋅P(¬A)

P(B)=(0.95⋅0.01)+(0.10⋅0.99)=0.1045

Now, apply Bayes's Theorem:

P(A∣B)≈0.0908

So, even though the test is relatively accurate, a person who tests positive still has only about an 9.08% chance of actually having the disease due to the low prevalence of the disease in the population.

This example demonstrates how Bayes's Theorem helps us update our beliefs based on new information, considering both the test's accuracy and the prior probability of the hypothesis being true.

#Q5
A confidence interval is a range of values that is likely to contain a population parameter, such as a mean or proportion, with a certain level of confidence. It is a way of quantifying the uncertainty of an estimate based on a sample of data.

The confidence interval is typically expressed as an interval estimate, such as "95% confidence interval for the mean is between 3.2 and 5.8". This means that if we were to repeat the sampling process many times, 95% of the time the true population mean would be within this range.

To calculate a confidence interval, we need to know the sample size, the sample mean, and the standard error of the mean (SEM), which is a measure of the variation in the sample means. The confidence interval is based on the standard normal distribution or t-distribution, depending on the sample size and whether the population standard deviation is known.

Here is an example of how to calculate a confidence interval:
Suppose we want to estimate the mean weight of a certain population of adults. We take a random sample of 100 people and measure their weights, and find that the sample mean is 72.5 kilograms and the sample standard deviation is 10.3 kilograms. We want to calculate a 99% confidence interval for the population mean weight.

Since the sample size is greater than 30 and the population standard deviation is unknown, we use the t-distribution. The degrees of freedom is n-1, which is 99 in this case.

First, we calculate the standard error of the mean:

SEM = s / sqrt(n) = 10.3 / sqrt(100) = 1.03

Next, we find the t-value for a 99% confidence level and 99 degrees of freedom using a t-table or calculator. The value is approximately 2.63.

The lower and upper bounds of the confidence interval are calculated as:

Lower bound = sample mean - t-value x SEM = 72.5 - 2.63 x 1.03 = 69.8
Upper bound = sample mean + t-value x SEM = 72.5 + 2.63 x 1.03 = 75.2
Therefore, the 99% confidence interval for the population mean weight is between 69.8 and 75.2 kilograms. This means that we are 99% confident that the true population mean weight lies within this range.

In [3]:
from math import sqrt
import scipy.stats as stats

sample_mean = 72.5
sample_std = 10.3
n = 100

SEM  = sample_std/sqrt(n)

alpha = 0.01

t_value = stats.t.ppf(1-alpha/2, df=n-1)
print(f't_value for {n} samples with significance level of {alpha} is : {t_value:.2f}')

lower_bound = sample_mean - t_value * SEM
upper_bound = sample_mean + t_value * SEM

print(f"The 99% confidence interval for population mean is ({lower_bound:.1f}, {upper_bound:.1f})")

t_value for 100 samples with significance level of 0.01 is : 2.63
The 99% confidence interval for population mean is (69.8, 75.2)


#Q6
Suppose you're investigating a rare genetic disorder (Event A) that occurs in approximately 1 in 100,000 individuals in a certain population. You have a genetic test that detects the disorder with 99% accuracy (sensitivity) when the individual actually has the disorder, and it correctly identifies individuals without the disorder (specificity) with a rate of 98%.

Now, you encounter an individual who has tested positive for the disorder using this genetic test. What is the probability that this individual actually has the disorder?

Given:

Prior probability of the disorder (P(A)): 1 in 100,000 (P(A)=0.00001).
Sensitivity of the test (P(B∣A)): 99% (P(B∣A)=0.99).
Specificity of the test (P(¬B∣¬A)): 98% (P(¬B∣¬A)=0.98).
Solution:
We want to find P(A∣B), which is the probability of the individual having the disorder given that they tested positive.

Using Bayes' Theorem:P(A∣B)= P(B∣A)⋅P(A)/P(B)

First, calculate 
P(B), the probability of testing positive:
P(B)=P(B∣A)⋅P(A)+P(B∣¬A)⋅P(¬A)

P(B)=(0.99⋅0.00001)+(0.02⋅0.99999)=0.0209798

Now, apply Bayes' Theorem:

P(A∣B)≈0.0004726

So, even though the individual tested positive for the disorder, the probability that they actually have the disorder is only about 0.04726%, which is much lower than the initial prior probability of 0.01%. This is because the test's accuracy, though high, is influenced by the low prevalence of the disorder in the population.



In [4]:
#Q7
import math
import scipy.stats as stats

sample_mean = 50
sample_std_dev = 5
sample_size = 30

alpha = 0.05

t_crit = stats.t.ppf(1-alpha/2, df=sample_size-1)
print(f't statistic for significance level {alpha} , sample size of {sample_size} is : {t_crit:.2f}')

std_error = sample_std_dev / math.sqrt(sample_size)

lower_bound = sample_mean - t_crit * std_error
upper_bound = sample_mean + t_crit * std_error

print(f"The 95% confidence interval is ({lower_bound:.2f}, {upper_bound:.2f})")

t statistic for significance level 0.05 , sample size of 30 is : 2.05
The 95% confidence interval is (48.13, 51.87)


#From above results we can say with 95% condfidence that the POPULATION MEAN Lies in between 48.13 and 51.87.

#Q8
The margin of error in a confidence interval is the range of values around the sample statistic (such as the sample mean) within which we expect the true population parameter (such as the population mean) to lie with a certain level of confidence.
The margin of error is affected by the sample size, level of confidence, and standard deviation of the population.
As the sample size increases, the margin of error decreases because larger samples provide more information about the population, and hence the sample statistics are more likely to be representative of the population.
The decrease in margin of error with an increase in sample size follows the square root law, which states that the margin of error decreases by the square root of the sample size.
For example, suppose we want to estimate the average height of all students in a university using a random sample of students. We take a sample of 50 students and calculate their average height to be 170 cm with a standard deviation of 5 cm. We want to calculate a 95% confidence interval for the true population mean height.

Using the formula for the margin of error for a confidence interval, we get:
Margin of error = z-value * (standard deviation / sqrt(sample size))
For a 95% confidence interval, the critical z-value is 1.96. So, plugging in the values, we get:
Margin of error = 1.96 * (5 / sqrt(50)) = 1.38 cm
This means that we are 95% confident that the true population mean height is within 1.38 cm of our sample mean of 170 cm. If we had taken a larger sample size of 200 students, the margin of error would have been:
Margin of error = 1.96 * (5 / sqrt(200)) = 0.98 cm
So, a larger sample size would result in a smaller margin of error, making our estimate of the true population mean height more precise.

In [6]:
#Q9
x_value = 75
u_value = 70 
sigma = 5 

# Calculate the z-score
z = (x_value - u_value) / sigma

print(f"The z-score is: {z}")

The z-score is: 1.0


Interpretation
This means that the data point is 1 standard deviation above the population mean.
A positive z-score indicates that the data point is above the mean, while a negative z-score indicates that it is below the mean.
The magnitude of the z-score tells us how far away the data point is from the mean in terms of standard deviations.

In [7]:
from scipy.stats import norm

percent_below = norm.cdf(z) * 100

print(f"The percentage of values below a z-score of 1 is: {percent_below:.2f}%")

The percentage of values below a z-score of 1 is: 84.13%


In [8]:
#Q10
import numpy as np
from scipy.stats import t

alpha = 0.05 
null_hypothesis = "the drug is not significantly effective"
alternative_hypothesis = "the drug is significantly effective"

mu = 0
sample_mean = 6
sample_std = 2.5
n = 50
df = n - 1

t_score = (sample_mean - mu) / (sample_std / np.sqrt(n))
p_value = 2 * (1 - t.cdf(abs(t_score), df))  # or stats.t.ppf(1 - alpha/2, dof)  # two-tailed t-value

print(f"t-statistic: {t_score:.4f}")
print(f"p-value: {p_value}")
if p_value < alpha:
    print(f"Reject the null hypothesis. {alternative_hypothesis}.")
else:
    print(f"Fail to reject the null hypothesis. {null_hypothesis}.")

t-statistic: 16.9706
p-value: 0.0
Reject the null hypothesis. the drug is significantly effective.


In [10]:
#Q11
import math

p = 0.65 
n = 500 
ci = 0.95
alpha = 1-ci
z_alpha = 1.96

se = math.sqrt((p * (1 - p)) / n)

me = z_alpha * se

lower_bound = p - me
upper_bound = p + me

print(f"The 95% confidence interval for the proportion of people who are satisfied with their job is ({lower_bound*100:.2f}%, {upper_bound*100:.2f}%)")
 

The 95% confidence interval for the proportion of people who are satisfied with their job is (60.82%, 69.18%)


In [11]:
#Q12
import numpy as np
from scipy.stats import t

x1 = 85
s1 = 6
n1 = 30

x2 = 82
s2 = 5
n2 = 30

alpha = 0.01
null_hypothesis = "mu1 = mu2 (the means are equal)"
alternative_hypothesis = "mu1 != mu2 (the means are not equal)"

sp = np.sqrt(((n1 - 1) * s1 ** 2 + (n2 - 1) * s2 ** 2) / (n1 + n2 - 2))

t_stat = (x1 - x2) / (sp * np.sqrt(1 / n1 + 1 / n2))

df = n1 + n2 - 2

t_crit = t.ppf(1-alpha / 2, df)

p_value = 2 * (1 - t.cdf(abs(t_stat), df))

print(f"t-statistic: {t_stat:.3f}")
print(f"Degrees of freedom: {df}")
print(f"Critical t-value: {t_crit:.3f}")
print(f"p-value: {p_value:.3f}")

if abs(t_stat) > t_crit:
    print(f"Since, t_stat > t_crit i.e. {t_stat:.3f} > {t_crit:.3f} : Reject null hypothesis.")
    print(alternative_hypothesis)
else:
    print(f"Since, t_stat < t_crit i.e. {t_stat:.3f} < {t_crit:.3f} : Fail to reject null hypothesis.")
    print(null_hypothesis)


t-statistic: 2.104
Degrees of freedom: 58
Critical t-value: 2.663
p-value: 0.040
Since, t_stat < t_crit i.e. 2.104 < 2.663 : Fail to reject null hypothesis.
mu1 = mu2 (the means are equal)


In [12]:
#Q13
import math
from scipy.stats import t

pop_mean = 60
pop_std = 8
n = 50
sample_mean = 65
confidence_level = 0.90
alpha = 1 - confidence_level # 1-0.90 = 0.1

t_score = (sample_mean - pop_mean) / (pop_std / math.sqrt(n))

df = n - 1
t_crit = t.ppf((1+confidence_level)/2, df) 
print(f't statistic for {n} samples with {confidence_level*100:.0f}% confidence is : {t_crit:.4f}')

margin_of_error = t_crit * (pop_std / math.sqrt(n))

lower_bound = sample_mean - margin_of_error
upper_bound = sample_mean + margin_of_error

print(f"The {confidence_level*100:.0f}% confidence interval for the true population mean is ({lower_bound:.3f}, {upper_bound:.3f})")

t statistic for 50 samples with 90% confidence is : 1.6766
The 90% confidence interval for the true population mean is (63.103, 66.897)


In [13]:
#Q14
import math
import scipy.stats as stats

null_hypothesis = "Caffeine has no significant effect on reaction time"
alternative_hypothesis = "Caffeine has a significant effect on reaction time"

sample_mean = 0.25 
sample_std_dev = 0.05  
n = 30 
pop_mean = 0.3 
alpha = 0.1 

t_stat = (sample_mean - pop_mean) / (sample_std_dev / math.sqrt(n))

t_crit = stats.t.ppf(1-alpha/2, df=n-1)

margin_of_error = t_crit * (sample_std_dev / math.sqrt(n))
lower_ci = sample_mean - margin_of_error
upper_ci = sample_mean + margin_of_error

print(f"t-statistic: {t_stat:.3f}")
print(f"t-critical value: {t_crit:.3f}")
print(f"90% Confidence Interval: ({lower_ci:.3f}, {upper_ci:.3f})")

if abs(t_stat) > t_crit:
    print("Reject the null hypothesis")
    print(alternative_hypothesis)
else:
    print("Fail to reject the null hypothesis")
    print(null_hypothesis)


t-statistic: -5.477
t-critical value: 1.699
90% Confidence Interval: (0.234, 0.266)
Reject the null hypothesis
Caffeine has a significant effect on reaction time
